### DISCUSSION TITLE
Controlling TLS extensions via Configuration file

### DISCUSSION THREAD

**Discussion by ayushiays** [openssl/openssl#25180] [Q&A]
I am looking to control (enable/disable) the following TLS extensions via Options flag as suggested in the doc https://docs.openssl.org/master/man3/SSL_CONF_cmd/#supported-configuration-file-commands but can not find the corresponding parameters.

    status_request
    status_request_v2
    psk_key_exchange_modes
    pre_shared_key
    early_data
    certificate_authorities
    ec_point_formats

I can only figure out the corresponding parameter for session_ticket which is SessionTicket but unable to do so for the other extensions mentioned above.
Can someone please point me to the right link.

**Comment by Sashan** [comment]
Let's convert it to discussion for now.

**Comment by mattcaswell** [comment]
- status_request

This extension cannot be controlled via the config file. It is via API calls only.

- status_request_v2

OpenSSL does not support this extension.

- psk_key_exchange_modes

This is controlled via the `AllowNoDHEKEX` and `PreferNoDHEKEX` options as described in the link you provided above.

- pre_shared_key

This cannot be controlled by the config file. It is via API calls only.

- early_data

This cannot be controlled by the config file. It is via API calls only.

- certificate_authorities

This can be controlled via the `RequestCAFile` configuration command as described in the link you provided above.

- ec_point_formats

This cannot be controlled by the config file directly. It's presence or absence can be controlled indirectly via other options (but not the contents of the ec_point_formats itself). If there are no EC capable ciphersuites or EC support is not built-in to OpenSSL then the extension is not sent.
